src/Service/Terminal/Service_Terminal.re

```diff
@@ -67,22 +67,30 @@ module Sub = {
         let makeDebouncedDispatch = () => {
           let scheduled = ref(false);
           let latestAction = ref(None);
+          let debounceCount = ref(0);
```

Comment:
This seems like a very imperative and error-prone way of doing this. I'm pretty sure a single `state` variant would be sufficient instead of multiple mutable variables that can be in any number of invalid states.

Reply:
Good point - working on a refactor